### DISCUSSION TITLE
Return table with lua functions

### DISCUSSION THREAD

**Discussion by SirEndii** [cc-tweaked/CC-Tweaked#1166] [Q&A] [answered]
Heyo,
I currently work on an ar goggles rework and I have a question
Is it possible to return a table with lua functions? Basically, we want that the user can create panels and control these panels individually.

As example
```lua
controller.mainPanel().addCircleElement(*args*)
--or
controller.mainPanel().addPanel("panelOne", *args*).addCircleElement
--or
controller.getPanel("panelOne").removeCircleElement

--No final names, but I hope you understand what I mean
```

**Comment by SquidDev** [comment, marked as answer]
https://github.com/cc-tweaked/CC-Tweaked/blob/db8c979a0626aebcedef90befdd1a9dd9eaf0add/src/main/java/dan200/computercraft/api/lua/MethodResult.java#L63-L64

I assume this is what you want - you should just be able to create a class for panels (or whatever) and add `@LuaFunction`s to it:

```java
class PanelElement {
  @LuaFunction
  public final PanelElement addPanel(String name) {
    return new PanelElement();
  }
}
```

**Reply by SirEndii** [reply]
Neat, thank you
